### PR TITLE
feat(logo-grid): add logoCount and logoRatio props

### LINKED ISSUE #8848: [Logo Grid]: Add design modifications to the "Grid logo" component
**Engineering info:**  
- Innovation Team engineer: @andy-blum 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2473
- DPO consulting engineer:

-----

### The problem

- This "Layout Component" is part of the Product Use Cases v1 Template (see EPIC - [AEMSITES-94](https://jsw.ibm.com/browse/AEMSITES-94))
- Figma: https://www.figma.com/file/EYcGXWm9O71Ma97MT1yflL/Components?node-id=2%3A63 
- PDF: https://ibm.ent.box.com/file/953618282844 

Pattern - 16_9 (1)

### The solution

- Add an attribute/prop for `logo-count`
  - 3 column (default)
  - 4 column
- Add an attribute/prop (eg. `logo-ratio`) for the different image ratios that we're supporting. The container of the logo item should be adjusted based on the defined ratio (see specs).

@oliviaflory @kennylam Feel free to add/adjust this solution area.

### Application/website

AEM

### Business priority

Medium Priority = upcoming release but is not pressing

### What time frame would this ideally be needed by (if applicable)

We'd like to get these modifications in for the v.1.35.0 release.

### Examples

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Closes #8848

### Description

Adds `logoCount` and `logoRatio` props to `DDSLogoGrid`

`logoCount` takes an integer to use as a number of columns for the grid. The stylesheet will constrain that value to between 2 and 4, inclusively.

`logoRatio` takes a string matching the format `<integer><separator><integer>` and sets a custom CSS property in an inline style to use in logo-grid-item elements as an aspect ratio.

### Changelog

**New**

- Added `logoCount` and `logoRatio` props to `DDSLogoGrid`

### CODE REVIEW

**Comment by oliviaflory** [comment]
Thanks for this @andy-blum!

### 1. Ratio in heading title
Storybook has been having some trouble with updating content with the knobs lately. The 4:3 is remaining when we change the ratio unless you open the story in its own tab, see the example of 2:1 but still says "4:3". I'd recommend removing the "4:3" for now and we could add it back when we figure out that weird storybook bug.
Screen Shot 2022-05-24 at 3 57 02 PM

### 2. Red border around images
Is this temporary just for testing? Our storybook demos usually show components as intended to be used, and I'd love to hide the red border so not to cause any confusion.

### 3. Remove `2` option within Column count
I don't believe @jwitkowski79 and I discussed a 2 up logo, so we'd like to remove that option and keep it at the 3 and 4 options.
In the specs Jen provided I do see a 2 column minimum outlined, but I believe that means a 2 column minimum width for the logos. Could you confirm @jwitkowski79 ?

### 4.  Medium breakpoint should have 4 logos across a row, not 3
The specs for the 4 column count option shows the logos spanning 4 across medium breakpoint. 
Screen Shot 2022-05-24 at 4 08 53 PM

### 5. Small breakpoint should have 2 logos across a row, not 1
The specs for the 4 column count option shows the logos spanning 2 across medium breakpoint. 
Screen Shot 2022-05-24 at 4 12 09 PM

### 6. Space between CTA and logos
Looks like somewhere we lost the 48px between the logo and CTA. We don't have to fix here, but I think that's what Jen is expecting eventually!

**Comment by jwitkowski79** [comment]
@oliviaflory @andy-blum Olivia is correct in saying that the logos should take up a 2 column count, and  2up logo variation needs to be removed.

Below is the proper columns-per-breakpoint configuration based on the design documentation:`<table role="table">
<thead>
<tr>
<th></th>
<th>logoCount: 3</th>
<th>logoCount: 4</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="user-mention notranslate" data-hovercard-type="organization" data-hovercard-url="/orgs/sm/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/sm">@sm</a></td>
<td>2 columns</td>
<td>2 columns</td>
</tr>
<tr>
<td><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/md/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/md">@md</a></td>
<td>2 columns</td>
<td>4 columns</td>
</tr>
<tr>
<td><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/lg/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/lg">@lg</a></td>
<td>3 columns</td>
<td>4 columns</td>
</tr>
</tbody>
</table>`
Also, not sure if this is in another upcoming ticket but we'd love to have the content block (not just the title) display above the the logo grid as seen in the design specs

File: packages/utilities/src/utilities/parseAspectRatio/parseAspectRatio.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @example
+ * import {parseAspectRatio} from '@carbon/ibmdotcom-utilities';
+ *
+ * const [width, height] = parseAspectRatio('16x9');
+ *
+ * @param {string} aspectRatioString string in format <integer><separator><integer>
+ * @returns {Array} [widthInt, heightInt]
+ */
+
+function parseAspectRatio(aspectRatioString) {
```
Comment (andy-blum) [inline packages/utilities/src/utilities/parseAspectRatio/parseAspectRatio.js:20]: CC @proeung @kennylam adding this utility as we'll utilize it in #8849

**Comment by andy-blum** [comment]
Not sure why the tests are failing - they pass for me locally. I'll close/re-open the PR to make the tests re-run.

File: packages/styles/scss/components/logo-grid/_logo-grid.scss [hunk 2]
```diff
@@ -1,10 +1,12 @@
 /**
- * Copyright IBM Corp. 2016, 2021
+ * Copyright IBM Corp. 2016, 2022
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
 
+/* stylelint-disable */
```
Comment (kennylam) [inline packages/styles/scss/components/logo-grid/_logo-grid.scss:8]: Why are we disabling this entire block?

**Review by kennylam — COMMENTED** [review]
Looks good @andy-blum, just a few observations.

**Comment by kennylam** [comment]
@andy-blum I think you only need to adjust the e2e tests. https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/6631472191?check_suite_focus=true#step:8:2339

**Comment by oliviaflory** [comment]
@andy-blum Thank you for the updates!

Small thing on the storybook knobs:

**1. Column count: Can we reduce to two options?** 
Because there isn't a difference between Default and Three, I'd love to go with `Three` (as the default) and `Four`
But if that would break anything in terms of renaming we can go with `Default` and `Four`
Screen Shot 2022-06-01 at 3 25 35 PM


**2. Column count: Can we reduce to three options?** 
Because there isn't a difference between Default and 4:3 I'd love to go with `4:3`, `16:9` and `2:1`
But if that would break anything in terms of renaming we can go with `Default`, `16:9` and `2:1`
Screen Shot 2022-06-01 at 3 25 54 PM

**Review by oliviaflory — APPROVED** [review]
Thanks @andy-blum !

**Review by proeung — APPROVED** [review]
Looks good! Thanks for addressing my feedback.

@jwitkowski79 Can you take a look at this PR and give us a thumbs up before we mark this one as ready for merge?

**Review by jwitkowski79 — APPROVED** [review]
@andy-blum @proeung This looks good!

**Pull request merged**